tempesta_fw/connection.h

```diff
@@ -284,6 +289,55 @@ tfw_srv_conn_hasnip(TfwSrvConn *srv_conn)
 	return test_bit(TFW_CONN_B_HASNIP, &srv_conn->flags);
 }
 
+/*
+ * Tell if connection is temporary inactive due to full work queue.
+ */
+static inline bool
+tfw_srv_conn_busy(TfwSrvConn *conn)
+{
+	if (test_bit(TFW_CONN_B_BUSY, &conn->flags))
+		return true;
+	rmb();
+	if (time_is_after_jiffies(READ_ONCE(conn->jbusytstamp)))
+		return true;
+
+	return false;
+}
+
+/*
+ * Make connection temporary inactive due to full work queue.
+ */
+static inline void
+tfw_srv_conn_set_busy(TfwSrvConn *conn)
+{
+	if (!test_bit(TFW_CONN_B_BUSY, &conn->flags))
+		set_bit(TFW_CONN_B_BUSY, &conn->flags);
```

Comment:
Why test the bit first when the goal is to just set it? There's nothing here that depends on the bit NOT set before. Also, this sequence doesn't even provide this guarantee as it's not atomic.

Reply:
Yes, you are right. Forgot to remove that redundant check after code changes.
Corrected.